### ISSUE TITLE
backup: network errors should be retried

### ISSUE THREAD

**Issue by mjibson** [cockroachdb/cockroach#14044]
When running backup on cobalt (two times) got separate errors (machine running on azure and saving to azure):

```
pq: exporting 3 ranges: first of 1 errors: storage: service returned error: StatusCode=500, ErrorCode=OperationTimedOut, ErrorMessage=Operation could not be completed within the specified time.
RequestId:0c0375ea-0001-0036-1a4e-98651d000000
Time:2017-03-08T20:55:32.1657120Z, RequestId=0c0375ea-0001-0036-1a4e-98651d000000, QueryParameterName=, QueryParameterValue=
```

and

```
pq: exporting 3 ranges: first of 1 errors: Put https://cockroachbackuptest.blob.core.windows.net/backup-test/cobalt-test-1/225895139076014082.sst?blockid=MQ%3D%3D&comp=block: write tcp 192.168.1.5:48298->13.68.165.72:443: write: connection reset by peer
```

Seems like we need some retries in there.

**Comment by dt** [comment]
this works for now, while we write the file before uploading, but won't be as straightforward when we stream uploads, so we'll need to retry the whole cmd Eval anyway.

**Comment by danhhz** [comment]
I think we need to leave that for when it happens. i'm unconvinced that we'll be streaming anytime soon

**Comment by mjibson** [comment]
This appears to have been fixed by #14109
